r/vignettes/developers/install_details.Rmd

```diff
@@ -0,0 +1,107 @@
+
+The diagram below shows how the R package finds a libarrow installation.
+
+```{r, echo=FALSE, out.width="70%"}
+knitr::include_graphics("../img/install_diagram.png")
+```
+
+### Using pkg-config
+
+When you install the arrow R package on Linux, if no environment variables 
+relating to the location of an existing libarrow installation have already by 
+set, the installation code will attempt to find libarrow on
+your system using the `pkg-config` command.
+
+This will find either installed system packages or libraries you've built yourself.
+In order for `install.packages("arrow")` to work with these system packages,
+you'll need to install them before installing the R package.
+
+### Prebuilt binaries
+
+If libarrow is not found on the system, the R package installation 
+script will next attempt to download prebuilt libarrow binaries
+that match your both your local operating system and arrow R package version.
+The libarrow binaries will only be retrieved if you have set the environment variable
+`LIBARROW_BINARY` or `NOT_CRAN`.
+
+If found, they will be downloaded and bundled when your R package compiles.
+For a list of supported distributions and versions,
+see the [arrow-r-nightly](https://github.com/ursa-labs/arrow-r-nightly/blob/master/README.md) project.
+
+### Building from source
+
+If no libarrow binary is found, it will attempt to build it locally.
+First, it will also look to see if you are in a checkout of the `apache/arrow` 
+git repository and thus have the libarrow source files there.
+Otherwise, it builds from the source files included in the package.
+Depending on your system, building libarrow from source may be slow. If 
+libarrow is built from source, `inst/build_arrow_static.sh` is executed.
+
+## Using the R package with libarrow installed as a system package
+
+If you are authorized to install system packages and you're installing a CRAN release,
+you may want to use the official Apache Arrow release packages corresponding to 
+the R package version via software distribution tools such as `apt` or `yum` 
+(though there are some drawbacks: see the 
+["Troubleshooting" section in the main installation docs]("../install.html)).
+See the [Arrow project installation page](https://arrow.apache.org/install/)
+to find pre-compiled binary packages for some common Linux distributions,
+including Debian, Ubuntu, and CentOS.
+
+Generally, we do not recommend this method of working with libarrow with the R 
+package unless you have a specific reason to do so. 
+
+## Using the R package with an existing libarrow build
+
+This setup is much more common for arrow developers, who may be needing to make changes to both the R package and libarrow source code.  See the [developer setup docs](./setup.html) for more information.
+
+# libarrow dependencies
+
+You'll need to install `libparquet-dev` on Debian and Ubuntu, or `parquet-devel` on CentOS.
+This will also automatically install libarrow as a dependency.
```

Comment:
This doesn't fit here but I'm not sure where it should go.

Reply:
Yeah I agree. Maybe we could [include the openssl dependencies](https://arrow.apache.org/docs/r/articles/install.html#s3-support) here too as "general system dependencies"? I think it's totally ok to duplicate in this case

Reply:
I don't think I understand "This will also automatically install libarrow as a dependency." - it sounds like it's saying "libarrow requires these things, which in turn, require libarrow", which can't be what it means, right?